File: app/src/androidTest/java/com/johnnyzhou/movieme/MainActivityInstrumentationTest.java
package com.johnnyzhou.movieme;

import android.support.test.espresso.action.ViewActions;
import android.support.test.rule.ActivityTestRule;
import android.support.test.runner.AndroidJUnit4;
import android.test.suitebuilder.annotation.LargeTest;
import android.widget.EditText;

import com.johnnyzhou.movieme.ui.MainActivity;

import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

import static android.support.test.espresso.Espresso.onView;
import static android.support.test.espresso.action.ViewActions.typeText;
import static android.support.test.espresso.matcher.ViewMatchers.isAssignableFrom;
import static android.support.test.espresso.matcher.ViewMatchers.withId;

@RunWith(AndroidJUnit4.class)
@LargeTest
public class MainActivityInstrumentationTest {
    public static final String SEARCH_STRING = "Avengers";

    @Rule
    public ActivityTestRule<MainActivity> activityRule = new ActivityTestRule<>(MainActivity.class);

    @Test
    public void searchAvengerMovie() {
        onView(withId(R.id.action_search)).perform(ViewActions.click());

        // Used to find the searchView edit text area
        onView(isAssignableFrom(EditText.class)).perform(typeText(SEARCH_STRING + "\n"));
    }
}

File: app/src/androidTest/java/com/johnnyzhou/movieme/MainActivityInstrumentationTest.kt
package com.johnnyzhou.movieme

import androidx.lifecycle.Lifecycle
import androidx.test.ext.junit.rules.ActivityScenarioRule
import androidx.test.ext.junit.runners.AndroidJUnit4
import androidx.test.filters.LargeTest
import com.johnnyzhou.movieme.annotations.OverrideLocalDate
import com.johnnyzhou.movieme.ui.MainActivity
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
@LargeTest
class MainActivityInstrumentationTest {
    @get:Rule
    var activityRule = ActivityScenarioRule(MainActivity::class.java)

    @get:Rule
    val localDateRule = OverrideLocalDateRule(year = 3233, month = 10, day = 13)

//    @get:Rule
//    val chain: RuleChain = RuleChain.outerRule(localDateRule)

    @Test
    fun searchAvengerMovie() {
        val scenario = activityRule.scenario
        scenario.moveToState(Lifecycle.State.RESUMED)
//        onView(withId(R.id.action_search)).perform(ViewActions.click())
    }
}
